test/execution-errors.js

```diff
@@ -1,83 +1,90 @@
+var exec = require('child_process').exec;
 var path = require('path');
 var os = require('os');
+
 var tildify = require('../lib/shared/tildify');
 
+var eraseTime = require('./tool/erase-time');
+var eraseLapse = require('./tool/erase-lapse');
+var sliceLines = require('./tool/slice-lines');
+var gulp = require('./tool/gulp-cmd');
+
 describe('execution error', function() {
 
   it('should output an error if a task is not defined', function(done) {
-    runner({ verbose: false })
-      .chdir('test/fixtures/gulpfiles')
-      .gulp('a')
-      .run(function(err, stdout, stderr) {
-        expect(err).toNotEqual(null);
-        expect(err.code).toEqual(1);
-        expect(eraseTime(stdout)).toMatch('Using gulpfile ');
-        expect(eraseTime(stderr)).toEqual(
-          'Task never defined: a\n' +
-          'To list available tasks, try running: gulp --tasks\n');
-        done();
-      });
+    var opts = { cwd: path.join(__dirname, './fixtures/gulpfiles') };
+    exec(gulp('a'), opts, cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).not.toBeNull();
+      expect(err.code).toEqual(1);
+      expect(eraseTime(stdout)).toMatch('Using gulpfile ');
+      expect(eraseTime(stderr)).toEqual(
+        'Task never defined: a\n' +
+        'To list available tasks, try running: gulp --tasks\n');
+      done();
+    }
   });
 
   it('should output an error if gulp version is unsupported', function(done) {
-    runner({ verbose: false })
-      .chdir('test/fixtures/errors/bad-gulp-version')
-      .gulp()
-      .run(function(err, stdout, stderr) {
-        expect(err).toNotEqual(null);
-        expect(err.code).toEqual(1);
-        expect(eraseTime(stdout)).toEqual('');
-        expect(eraseTime(stderr)).toEqual('Unsupported gulp version\n');
-        done();
-      });
+    var opts = { cwd: path.join(__dirname, './fixtures/errors/bad-gulp-version') };
+    exec(gulp(), opts, cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).not.toBeNull();
+      expect(err.code).toEqual(1);
+      expect(eraseTime(stdout)).toEqual('');
+      expect(eraseTime(stderr)).toEqual('Unsupported gulp version 1.2.3\n');
+      done();
+    }
   });
 
   it('should output an error if gulp is not found', function(done) {
-    runner({ verbose: false })
-      .chdir(os.tmpdir())
-      .gulp()
-      .run(function(err, stdout, stderr) {
-        expect(err).toNotEqual(null);
-        expect(err.code).toEqual(1);
-        expect(eraseTime(stdout)).toEqual('');
-        stderr = eraseTime(stderr).split(/[\r\n]+/);
-        expect(stderr[0]).toMatch('Local gulp not found in ');
-        expect(stderr[1]).toEqual('Try running: npm install gulp');
-        done();
-      });
+    var tmpdir = os.tmpdir();
+    if (os.platform() === 'win32') {
+      var moveDrive = tmpdir.slice(0, 2);
+      exec(moveDrive + '& cd ' + tmpdir + ' & ' + gulp(), cb);
+    } else {
+      exec(gulp(), { cwd: tmpdir }, cb);
+    }
```

Comment:
Do we still need this `moveDrive` logic if we are using `cwd`? Not going to block on it

Reply:
#247